projects/core/src/lib/components/legend/legend-layer/legend-layer.component.ts

```diff
@@ -7,14 +7,44 @@ import { AppLayerModel } from '@tailormap-viewer/api';
   styleUrls: ['./legend-layer.component.css'],
   changeDetection: ChangeDetectionStrategy.OnPush,
 })
-export class LegendLayerComponent {
+export class LegendLayerComponent implements OnChanges {
 
   @Input()
   public layer: AppLayerModel | null = null;
 
   @Input()
   public url: string | null = null;
 
+  public urlWithOptions: string | null = null;
+
+  public srcset = '';
+
   public failedToLoadMessage = $localize `Failed to load legend for`;
 
+  public ngOnChanges() {
+
+    this.urlWithOptions = this.url;
+    this.srcset = '';
+
+    if (this.url) {
+      const url = new URL(this.url);
+      if (url.pathname.includes('/geoserver/') && url.searchParams.get('REQUEST') === 'GetLegendGraphic') {
```

Comment:
Why does this not use the [`ServerTypeHelper`](https://github.com/B3Partners/tailormap-viewer/blob/main/projects/map/src/lib/helpers/server-type.helper.ts) ?

If at some point we start communicating the server type from the backend this is likely to be left out/ignored

Reply:
Good catch, the server type is already sent but only in the `ServiceModel` but we only have the `AppLayerModel` from the parent component atm.

See other WIP where I do use it: https://github.com/B3Partners/tailormap-viewer/pull/144/files#diff-3fac1d7cb85a7f44a62027cc4c5f9c22aff9cc83d70c050a844cd4352c7b4f7eR116